sklearn/pipeline.py

```diff
@@ -2113,6 +2114,15 @@ def get_metadata_routing(self):
 
         return router
 
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags.input_tags.sparse = all(
+            get_tags(trans).input_tags.sparse
+            for name, trans in self.transformer_list
+            if trans not in {"passthrough", "drop"}
+        )
+        return tags
+
```

Comment:
@glemaitre should I be defensive against invalid `transformers_list` with a `try/except` as is done for `Pipeline` and `_BaseHeterogeneousEnsemble` ?

Reply:
The same holds for `ColumnTransformer` and `self.transformers`.

Reply:
Yep because you can end-up in this function before to have call the `fit` function that is in charge of validating the `self.transformer_list`.

Reply:
done in 21ec585